.github/workflows/run-tests.yml

```
@@ -0,0 +1,36 @@
+name: Run Unit Tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: '*'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10  
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10"]
+      fail-fast: false  
+
+    steps:
+    - uses: actions/checkout@v3
+    
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
```

Similarly, use latest version (unless you have this due to compatibility issues):
```suggestion
      uses: actions/setup-python@v5
```